runelite-client/src/main/java/net/runelite/client/plugins/bank/BankItemsOverlay.java

```diff
@@ -0,0 +1,166 @@
+import java.util.Arrays;
+import java.util.Comparator;
+
+@Slf4j
+class BankItemsOverlay extends Overlay
+{
+	private static final int ITEM_SIZE = 10;
+	private static final int PLACEHOLDER_WIDTH = 36;
+	private static final int PLACEHOLDER_HEIGHT = 32;
+	private static final int WRAPPING = 2;
+	private static final Point GAP = new Point(10, 6);
+	private static final ImageComponent PLACEHOLDER_IMAGE = new ImageComponent(new BufferedImage(PLACEHOLDER_WIDTH, PLACEHOLDER_HEIGHT, BufferedImage.TYPE_4BYTE_ABGR));
+
+	private final PanelComponent panelComponent = new PanelComponent();
+
+	private final ItemManager itemManager;
+	private final BankConfig config;
+	private final BankCalculation bankCalculation;
+
+
+	private ItemPrice[] expensiveItems;
+
+	@Inject
+	private BankItemsOverlay(ItemManager itemManager, BankConfig config, BankCalculation bankCalculation)
+	{
+		setPosition(OverlayPosition.BOTTOM_RIGHT);
+		panelComponent.setWrapping(WRAPPING);
+		panelComponent.setGap(GAP);
+		panelComponent.setOrientation(ComponentOrientation.HORIZONTAL);
+		panelComponent.setPreferredSize(new Dimension(PLACEHOLDER_WIDTH + GAP.x, PLACEHOLDER_HEIGHT + GAP.y));
+
+		this.itemManager = itemManager;
+		this.config = config;
+		this.bankCalculation = bankCalculation;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showMostExpensiveItemsOverlay())
+		{
+			return null;
+		}
+
+		final Item[] bankItems = bankCalculation.getBankItems();
+
+		if (bankItems == null || bankItems.length == 0)
+		{
+			return null;
+		}
+
+		panelComponent.getChildren().clear();
+
+		if (expensiveItems == null || bankCalculation.isBankDifferent(bankItems)) 
+		{
+			expensiveItems = getExpensiveItems(bankItems);
+			Arrays.sort(expensiveItems, Comparator.comparing(ItemPrice::getPrice).reversed());
+		}
+
+
+		for (int i = 0; i < ITEM_SIZE; i++)
+		{
+			if (i < expensiveItems.length)
+			{
+				final ItemPrice itemPrice = expensiveItems[i];
+				if (itemPrice.getItem().getQuantity() > 0)
+				{
+					final BufferedImage image = getImage(itemPrice.getItem());
+
+					if (image != null)
+					{
+						ImageComponent imageComponent = new ImageComponent(image);
+						panelComponent.getChildren().add(imageComponent);
+
+						TitleComponent titleComponent = TitleComponent.builder()
+								.color(Color.white)
+								.text(StackFormatter.quantityToStackSize(itemPrice.getPrice()))
+								.build();
+						panelComponent.getChildren().add(titleComponent);
+						continue;
+					}
+				}
+			}
+
+			// put two placeholder components (one image, one text) so the panel is not resized
+			panelComponent.getChildren().add(PLACEHOLDER_IMAGE);
+			panelComponent.getChildren().add(TitleComponent.builder().text("").build());
+		}
+
+		return panelComponent.render(graphics);
+
+	}
+
+	private BufferedImage getImage(Item item)
+	{
+		return itemManager.getImage(item.getId(), item.getQuantity(), true);
+	}
+
+	private ItemPrice[] getExpensiveItems(Item[] items)
```

Comment:
This should be calculated in the plugin.

Reply:
Why is that? This is the only place it's used.

Reply:
Because you're running this one per frame in the overlay, when it only needs to be run once each time the bank is updated.

Reply:
FWIW, it *is* run only each time the bank is updated

Reply:
You're right, though it is comparing the bank contents to the previously stored bank contents every frame the bank is open.
The point is, raiyni suggested this be tracked and calculated in the plugin because you then have access to `@Subscribe`r methods for event-based programming, rather than being fully declarative in this overlay. That would have the benefit of running this code (and the bank comparison) exactly one time upon changes instead of 50x/second.